src/main/java/Ui.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * This is the Ui class. It makes sense of what the input is.
+ */
+
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Scanner;
+public class Ui {
+    /**
+     * <p>
+     *     run is used to start up duke.
+     * </p>
+     * @param sc input command
+     * @param array the list of task
+     * @param num start with 1 for first task
+     */
+  
+    public void run(Scanner sc, ArrayList<TaskList> array, int num) {
+        Parser p = new Parser();
+        while (sc.hasNext()) {
+            String word = sc.nextLine();
+
+            while (!word.equals("bye") && !word.equals("list") && !
```

Comment:
if you convert everything into a switch statement with default being the error, then you don't need the loops